### PR TITLE
Move `ApiClient.getPOIs()` endpoints to Retrofit-based API client

### PR DESCRIPTION
**Pull request by oliverlockwood**
This is the first step in moving towards addressing https://github.com/cyclestreets/android/issues/133, by iteratively moving away from any dependency on the Apache HttpClient.

Key points worth highlighting:
-  By leveraging Retrofit/OKHttp/Gson, I believe the use of the underlying CycleStreets APIs becomes much clearer to the developer/maintainer; we also leave JSON parsing to a library, enabling us to remove a whole bunch of code.  (Note that we could equally use Jackson or similar for JSON conversion; Retrofit's documentation recommended Gson in the first instance so I went with that.)
-  For now, I am leaving the interface to the rest of the app as the static `ApiClient` class.  Once full migration is complete, we may want to revisit that, but I believe it makes sense to change things in small chunks.
-  Perhaps controversially given their previous absence 😉 , I've included unit tests of the new functionality. (Ideally, I would have liked to first create working unit tests of the old `ApiClient`, but it turned out that I'm not currently familiar enough with Android's intricacies to figure out how to get that working in a reasonable timescale.)

@jezhiggins @mvl22 This is ready for review when you have the time.  I'll hold off migrating the other `ApiClient` operations until the strategy has been fully agreed, to save us from rework.

### CODE REVIEW

**Comment by mvl22** [comment]
Thanks, Oliver.

I don't know if this adds any value, or is worthwhile considering, but it might be worth seeing if there is a standard GeoJSON parsing library available, as all the API V2 output that is geographical (i.e. almost everything) will be in the standard GeoJSON layout - I notice all the manual definition of types like lat/lon etc, which presumably a library might do automatically.

In the case of Leaflet.js for the web, we just load the GeoJSON parser and it basically knows how to connect everything up automatically.

File: libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/RetrofitApiClient.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RetrofitApiClient {
+
+  private final V1Api v1Api;
+  private final V2Api v2Api;
+
+  public RetrofitApiClient(String v1Host, String v2Host, String apiKey) {
```
Comment (mvl22) [inline libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/RetrofitApiClient.java]: I would set the order as `v2Host, v1Host, apiKey` or `v2Host, apiKey, v1Host` as V1 is basically deprecated and the least important of the three.

Also, the V1 API does run over HTTPS if you want, and there's no reason not to do that, so please update references from http://www.cyclestreets.net/ to https://www.cyclestreets.net/

Comment (oliverlockwood) [inline libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/RetrofitApiClient.java]: Actually, I'd go one further and say that having a constructor with three `String` parameters is a bad smell.  Consequently, I've changed this to use a builder... see what you think.

Regarding updating http references to https, are you referring just to the `ApiClient` itself (https://github.com/cyclestreets/android/blob/master/libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/ApiClient.java#L65) or more widely?  If the latter, I think it warrants its own PR separate from this one.

**Comment by oliverlockwood** [comment]
@mvl22 great call on using a library for GeoJSON parsing / POJO definitions;  I hadn't processed that the API response was standard GeoJSON, so thanks for the pointer.

PR updated.  With this in place we're able to remove some more boilerplate code, which is great.  Note that as a result of this I've switched to using Jackson instead of Gson with retrofit, as the GeoJSON library support for Jackson seemed to be better.

**Comment by mvl22** [comment]
Thanks again, Oliver.

@jezhiggins Can you review?

**Pull request merged**